pkg/storage/stores/shipper/compactor/deletion/delete_requests_table.go

```diff
@@ -183,7 +183,7 @@ func (t *deleteRequestsTable) BatchWrite(ctx context.Context, batch chunk.WriteB
 
 func (t *deleteRequestsTable) QueryPages(ctx context.Context, queries []chunk.IndexQuery, callback func(chunk.IndexQuery, chunk.ReadBatch) (shouldContinue bool)) error {
 	for _, query := range queries {
-		if err := t.boltdbIndexClient.QueryDB(ctx, t.db, query, callback); err != nil {
+		if err := t.boltdbIndexClient.QueryDB(ctx, t.db, nil, query, callback); err != nil {
```

Comment:
I'm curious your thoughts on calling on that the Pages query always operates against the index bucket by maybe changing that variable name and making it public? I thinks this adds a bit of clarity to the reader, rather than having to dig into the behavior to `QueryDB` to understand the behavior of a `nil` bucket name?

```suggestion
		if err := t.boltdbIndexClient.QueryDB(ctx, t.db, local.IndexBucketName, query, callback); err != nil {
```

Reply:
Sounds good! Will take care of it.